### ISSUE TITLE
Downloads are broken because of semicolon appended to filename

### ISSUE THREAD

**Issue by bbasinski** [phalcon/cphalcon#13919]
Changes from #13496  breaks our all downloads. Downloaded file has semicolon appended  to the extension which disable file type recognition by system (which is important). Tested both linux and windows, chrome and firefox. Even [RFC 6266  Content-Disposition in HTTP ](https://tools.ietf.org/html/rfc6266) omits semicolon at the end of header. Browser does not strip this. Tried to wrap filename in double quotes but with no luck. Only solution is to overwrite Content-Disposition header to remove semicolon from the end.

```php
$this->response->setFileToSend($fileDir, $filename);

//fix
$this->response->setHeader("Content-Disposition", "attachment; filename=$filename");
```

### Details

* Phalcon version:  3.4.2
* PHP Version: 7.2.15
* Operating System: Debian
* Server:  Apache

**Comment by StudioMaX** [comment]
I always hated the Content-Disposition header. The current implementation in Phalcon is suitable for the simplest cases - when the filename is without any encodings and spaces, like filename.txt. For more complex file names each browser will react differently. Just look at this Test case: http://test.greenbytes.de/tech/tc2231/
Personally I use something like this:
```php
if ($browserInfo['browser_name'] === 'Internet Explorer' && ($browserInfo['browser_version'] === '7.0' || $browserInfo['browser_version'] == '8.0')) {
    $filename = mb_convert_encoding($filename, 'UTF-8', 'Windows-1251');
    $contentDisposition = 'attachment; filename="' . rawurlencode($filename) . '"';
} elseif ($browserInfo['browser_name'] === 'Safari') {
    $filename = mb_convert_encoding($filename, 'UTF-8', 'Windows-1251');
    $contentDisposition = 'attachment; filename="' . $filename . '"';
} else {
    $filename = mb_convert_encoding($filename, 'UTF-8', 'Windows-1251');
    $contentDisposition = 'attachment; filename="' . $filename . '"; filename*=UTF-8\'\'' . rawurlencode($filename);
}

$this->response->setHeader("Content-Disposition", $contentDisposition);
```
Fortunately, Internet Explorer 7.0 and 8.0 are already dead, but Safari can still be a problem.

And yes, in any of these cases you don't need a semicolon at the end, but the filename must be enclosed in quotes.

**Comment by sergeyklay** [comment]
Fixed in the `4.0.x` branch. Feel free to open a new issue if the problem appears again. Thank you for the bug report.
